platforms/jvm/plugins-jvm-test-suite/src/main/java/org/gradle/api/plugins/JvmTestSuitePlugin.java

```diff
@@ -84,14 +83,15 @@ public void apply(Project project) {
                     .willBeRemovedInGradle9()
                     .withUpgradeGuideSection(8, "test_task_default_classpath")
                     .nagUser();
-                return JavaPluginHelper.getDefaultTestSuite(project).getSources().getOutput().getClassesDirs();
+
+                return ((JvmTestSuite) testing.getSuites().findByName(JvmTestSuitePlugin.DEFAULT_TEST_SUITE_NAME)).getSources().getOutput().getClassesDirs();
```

Comment:
This will break if the Java Plugin is not applied before the test suite plugin is applied (it was already broken since we used the Java Plugin helper without applying the Java Plugin)

It might make sense to verify `findByName` returns non null and also do an instanceof check

Reply:
ok